gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Client.java

```diff
@@ -511,19 +515,24 @@ protected Connection chooseConnection(final RequestMessage msg) throws TimeoutEx
          */
         @Override
         protected void initializeImplementation() {
+            // use a special executor here to initialize the Host instances as the worker thread pool may be
+            // insufficiently sized for this task and the parallel initialization of the ConnectionPool. if too small
+            // tasks may be schedule in such a way as to produce a deadlock: TINKERPOP-2550
+            //
+            // the cost of this single threaded executor here should be fairly small because it is only used once at
+            // initialization and shutdown. since users will typically construct a Client once for the life of their
+            // application there shouldn't be tons of thread pools being created and destroyed.
+            final BasicThreadFactory threadFactory = new BasicThreadFactory.Builder().namingPattern("gremlin-driver-init-%d").build();
+            final ExecutorService hostExecutor = Executors.newSingleThreadExecutor(threadFactory);
```

Comment:
starting and shutting down executor is an expensive operation which will impact the client start up/shut down time which will impact use cases when customers have short running clients such as in AWS Lambda. Is there an alternative approach that can avoid this?

Reply:
can we not take a few milliseconds to fix this in a simple way? in multiple tests of 3.4.10 vs this change, the longest time difference i've seen is like 3.5 milliseconds extra time per open/close and many times the gap is even less. too much still?